### PR TITLE
rely on options for passing node to base widget

### PR DESCRIPTION
**Pull request by sccolbert**

### CODE REVIEW

**Comment by blink1073** [comment]
LGTM.

File: src/ui/commandpalette.ts [hunk 1]
```diff
@@ -241,31 +241,10 @@ namespace CommandItem {
...
-    node.appendChild(content);
-    return node;
-  }
-
-  /**
    * Construct a new command palette.
    */
   constructor(options: CommandPalette.IOptions = {}) {
-    super();
+    super({ node: Private.createNode() });
```
Comment (afshin) [inline src/ui/commandpalette.ts]: Could you explain the rationale for why this isn't part of `IRenderer`, *i.e.*:
```typescript
super({ node: (options.renderer || CommandPalette.defaultRenderer).createNode() });
```

Comment (blink1073) [inline src/ui/commandpalette.ts]: Probably because the private code relies on this DOM structure

Comment (sccolbert) [inline src/ui/commandpalette.ts]: Exactly. The widget relies on the structure of the node. External code can customize the dynamic content, but not the static content.

Comment (afshin) [inline src/ui/commandpalette.ts]: Gotcha, thanks. Looks good to me 👍

**Pull request merged**